### DISCUSSION TITLE
TCE on Supabase Postgres database using pgsodium with Prisma?

### DISCUSSION THREAD

**Discussion by codentacos** [prisma/prisma#18563] [Q&A] [answered]
Supabase supports TCE(Transparent Column Encryption) for Postgres databases using `pgsodium` [Supabase article on TCE](https://supabase.com/blog/transparent-column-encryption-with-postgres#one-key-id-for-the-entire-column). 

Is there a way to implement TCE using `pgsodium` and `Prisma`? I haven't been able to find any examples or references of the two used together.

**Comment by nurul3101** [comment]
Hello @codentacos 👋 

It looks like `pgsodium` is a PostgreSQL extension for cryptography, If the extension is enabled on your database you should be able to use `pgsodium` with [raw queries](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access).

Can you try it? Let us know in case you run into any issues.

**Reply by codentacos** [reply]
Yea that was kinda my thought at one way it might be possible. I'm not quite sure where to utilize it however. Given the examples it seems the `SECURITY LABEL` gets applied right after the table is created but as far as I know you can't run raw queries in the schema file so where would the next best place to do this be?

Example from Supabase
```SQL
CREATE TABLE credit_cards (
	id bigserial primary key,
  	credit_card_number text,
	key_id uuid not null DEFAULT 'e348034b-3f07-4878-aad6-000511d12826'::uuid
);

SECURITY LABEL FOR pgsodium
	ON COLUMN credit_cards.credit_card_number
  	IS 'ENCRYPT WITH KEY COLUMN key_id';
```

**Reply by janpio** [reply, marked as answer]
You can use `npx prisma migrate dev --create-only` when initially migrating your schema and modify the generated `migrations.sql` file to add this SQL. Then you can deploy those migrations with `npx prisma migrate deploy`.

**Reply by nurul3101** [reply]
Here's a [reference](https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/customizing-migrations#how-to-edit-a-migration-file) for customizing your migration file using `--create-only` flag.
